### DISCUSSION TITLE
Need some help and guidance in ZBus error on Linux

### DISCUSSION THREAD

**Discussion by earljohn004** [hoodie/notify-rust#191] [Q&A]
Hello, 

I am new to rust and notify-rust. 
I am making some POC for some application and I want to integrate this. 

Problem: 
I am encountering some errors on Linux environment on runtime and I don't understand what is wrong on Linux. I have referred to toastify and this is my current configuration

```
[dependencies.notify-rust]
version = "4.5"
default-features = false

[features]
default = ["z"]
d = ["notify-rust/d"]
z = ["notify-rust/z"]
images = ["notify-rust/images"]
```

I have created some helper function in some crate. It looks just like this
```
use notify_rust::Notification;

pub fn show_toast_notification() -> Result<(), notify_rust::error::Error> {
    Notification::new()
        .summary("Test Summary")
        .body("This is a dummy error. Testing notification")
        .show()?;
    Ok(())
}
```

I need this to work in cross platform: 
- Mac ( working )
- Windows 10 ( working )
- Windows 11 ( working )
- Linux ( Not working )  - please see error below
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Zbus(InputOutput(Os { code: 2, kind: NotFound, message: "No such file or directory" })) }
```

I am confused why I am encountering an error in ZBus with "No such file or directory". Maybe you have an idea on what this could mean. 

Thank you.

**Comment by hoodie** [comment]
Hiho, the Error is indeed not very informative. What I've seen people forget in the past was that there needs to be an actual dbus bus and service running on your system. That is usually the case when you have a desktop environment like KDE, Gnome, Xfce and the like, but on a bare-bone no-ui linux like in a docker container etc that is usually not the case. Could that be the source of your issue?

**Reply by earljohn004** [reply]
Hello, 

Thanks for responding. Actually this issue occurs on a CentOS 8 environment with UI. 

I will get back on this and try to find the cause. I haven’t gotten back since.

**Comment by m00sp** [comment]
@earljohn004 your same example is working perfect here on alpine linux 3.20.2, i'm using kde plasma with dbus. The only modification needed in my case is set default to "d":
```
default = ["d"]
```
by the way thank you for helper funtion.
> Hello,
> 
> I am new to rust and notify-rust. I am making some POC for some application and I want to integrate this.
> 
> Problem: I am encountering some errors on Linux environment on runtime and I don't understand what is wrong on Linux. I have referred to toastify and this is my current configuration
> 
> ```
> [dependencies.notify-rust]
> version = "4.5"
> default-features = false
> 
> [features]
> default = ["z"]
> d = ["notify-rust/d"]
> z = ["notify-rust/z"]
> images = ["notify-rust/images"]
> ```
> 
> I have created some helper function in some crate. It looks just like this
> 
> ```
> use notify_rust::Notification;
> 
> pub fn show_toast_notification() -> Result<(), notify_rust::error::Error> {
>     Notification::new()
>         .summary("Test Summary")
>         .body("This is a dummy error. Testing notification")
>         .show()?;
>     Ok(())
> }
> ```
> 
> I need this to work in cross platform:
> 
>     * Mac ( working )
> 
>     * Windows 10 ( working )
> 
>     * Windows 11 ( working )
> 
>     * Linux ( Not working )  - please see error below
> 
> 
> ```
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Zbus(InputOutput(Os { code: 2, kind: NotFound, message: "No such file or directory" })) }
> ```
> 
> I am confused why I am encountering an error in ZBus with "No such file or directory". Maybe you have an idea on what this could mean.
> 
> Thank you.
